crates/aranya-client-capi/src/api/defs.rs

```diff
@@ -952,12 +980,240 @@ pub fn setup_default_roles(client: &mut Client, team: &TeamId) -> Result<(), imp
+    team: &TeamId,
+    role: &RoleId,
+    managing_role: &RoleId,
+    perm: Permission,
+) -> Result<(), imp::Error> {
+    // SAFETY: Caller must ensure `perm` is a valid C String.
+    let perm = unsafe { perm.as_underlying() }?;
+
+    client.rt.block_on(
+        client
+            .inner
+            .team(team.into())
+            .assign_role_management_permission(role.into(), managing_role.into(), perm),
+    )?;
+
+    Ok(())
+}
+
+/// Revokes a role management permission from a managing role.
+///
+/// @param[in] client the Aranya Client [`Client`].
+/// @param[in] team the team's ID [`TeamId`].
+/// @param[in] role the ID of the subject role [`RoleId`].
+/// @param[in] managing_role the ID of the managing role [`RoleId`].
+/// @param[in] perm the management permission to assign [`Permission`].
+///
+/// @relates AranyaClient.
+pub fn revoke_role_management_permission(
+    client: &Client,
+    team: &TeamId,
+    role: &RoleId,
+    managing_role: &RoleId,
+    perm: Permission,
+) -> Result<(), imp::Error> {
+    // SAFETY: Caller must ensure `perm` is a valid C String.
+    let perm = unsafe { perm.as_underlying() }?;
+
+    client.rt.block_on(
+        client
+            .inner
+            .team(team.into())
+            .revoke_role_management_permission(role.into(), managing_role.into(), perm),
+    )?;
+
+    Ok(())
+}
+
+/// Changes the `role` on a `device`
+///
+/// This will change the device's current role to the new role assigned.
+///
+/// Permission to perform this operation is checked against the Aranya policy.
+///
+/// @param[in] client the Aranya Client [`Client`].
+/// @param[in] team the team's ID [`TeamId`].
+/// @param[in] device the device's ID [`DeviceId`].
+/// @param[in] old_role the ID of the role currently assigned to the device.
+/// @param[in] new_role the ID of the role to assign to the device.
+///
+/// @relates AranyaClient.
+pub fn change_role(
+    client: &Client,
+    team: &TeamId,
+    device: &DeviceId,
+    old_role: &RoleId,
+    new_role: &RoleId,
+) -> Result<(), imp::Error> {
+    client
+        .rt
+        .block_on(client.inner.team(team.into()).change_role(
+            device.into(),
+            old_role.into(),
+            new_role.into(),
+        ))?;
+
+    Ok(())
+}
+
+/// Returns all of the roles for this team.
+///
+/// Returns an `AranyaBufferTooSmall` error if the output buffer is too small to hold the roles.
+/// Writes the number of roles that would have been returned to `roles_len`.
+/// The application can use `roles_len` to allocate a larger buffer.
+///
+/// @param[in] client the Aranya Client [`Client`].
+/// @param[in] team the team's ID [`TeamId`].
+/// @param[out] roles_out returns a list of roles on the team [`Role`].
+/// @param[in,out] roles_len the number of roles written to the buffer.
+///
+/// @relates AranyaClient.
+pub unsafe fn team_roles(
+    client: &Client,
+    team: &TeamId,
+    roles_out: *mut MaybeUninit<Role>,
+    roles_len: &mut usize,
```

Comment:
Either `roles_out` and `roles_out_len` or `roles` and `roles_len`
```suggestion
    roles: *mut MaybeUninit<Role>,
    roles_len: &mut usize,
```